qiskit/transpiler/passes/synthesis/unitary_synthesis.py

```diff
@@ -501,9 +502,27 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
             if plugin_method.supports_coupling_map or default_method.supports_coupling_map
             else {}
         )
-        return self._run_main_loop(
-            dag, qubit_indices, plugin_method, plugin_kwargs, default_method, default_kwargs
-        )
+
+        if self.method == "default" and isinstance(kwargs["target"], Target):
+            _coupling_edges = (
+                list(self._coupling_map.get_edges()) if self._coupling_map is not None else []
+            )
+
+            out = run_default_main_loop(
+                dag,
+                list(qubit_indices.values()),
```

Comment:
Is there a chance that you might not need these anymore? AFAIK in rust the `DAGCircuit` and `CircuitData` datamodels store their respective `Qubit`/`Clbit` indices already. I might be wrong on this but it would be worth taking a look.

Reply:
These indices are really only used when there are control flow instructions in the circuit. In these cases `py_run_main_loop` is called recursively over the blocks (L255) and the block dag ids may not match the original dag ids, which we need to access instruction properties in the target. So this attribute acts like a map we can use to retrieve the original ids.